### ISSUE TITLE
BottomNavigationbar is not save the current body state

### ISSUE THREAD

**Issue by iBotasky** [flutter-devs/fluter_bottom_navigation_demo#1]
```dart
return new Scaffold(
      appBar: AppBar(
        title: const Text("Gallery"),
        actions: <Widget>[
          new IconButton(
              icon: new Icon(Icons.search), onPressed: () {}),
          new IconButton(
              icon: new Icon(Icons.more_vert), onPressed: () {})
        ],
      ),
     body: viewContainer[currentIndex],
      bottomNavigationBar: new BottomNavigationBar(
                  onTap: onTabTapped, 
          currentIndex: currentIndex,
          fixedColor: Colors.red,
          items: [
            BottomNavigationBarItem(
          icon:new Icon(Icons.photo),
          title: Text("Photo"),
        ),
        BottomNavigationBarItem(
          icon:new Icon(Icons.video_library),
          title: Text("Video"),
        ),
        BottomNavigationBarItem(
            icon:new Icon(Icons.photo_library),
          title: Text("Album"),
        )
      ]),
    );
```
@flutter-devs Hello, I got a issue
The  list of `body` and Bottomnavigationbar has some issue, while I'm in the first container, and i scroll to the bottom of this container, and then I tap to the second bar turns to the second container, and tap back the first bar back to first container, the container will rebuild the first container(Maybe it's init the container again), the scroller will back to the top of the container. 
In short, the `body` will not save the view states of the container.
How to resolves it ?
